units/size.go

```diff
@@ -0,0 +1,101 @@
+/*
+Copyright (c) 2015 VMware, Inc. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package units
+
+import (
+	"errors"
+	"fmt"
+	"math"
+	"regexp"
+	"strconv"
+)
+
+type ByteSize float64
```

Comment:
Initially I was confused by this comment because the user may actually be more precise.

i.e.
`govc vm.disk.create -size=10.5g ...`
vs
`govc vm.disk.create -size=10g ...`

But I believe you are referring the floating point arithmetic of int64(size.0 / 1024.0) possibly losing precision. Correct? If so, then I agree it makes sense to use int64.